File: src/main/java/br/com/alura/forum/repository/OpenTopicByCategoryRepository.java
package br.com.alura.forum.repository;

import java.util.List;

import br.com.alura.forum.model.OpenTopicByCategory;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.Repository;

public interface OpenTopicByCategoryRepository extends Repository<OpenTopicByCategory, Long> {
	void saveAll(Iterable<OpenTopicByCategory> topics);

	@Query("select t from OpenTopicByCategory t " +
            "where year(t.date) = year(current_date) " +
            "and month(t.date) = month(current_date)")
	List<OpenTopicByCategory> findAllByCurrentMonth();
}

File: src/main/kotlin/br/com/alura/forum/repository/OpenTopicByCategoryRepository.kt
package br.com.alura.forum.repository

import br.com.alura.forum.model.OpenTopicByCategory
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.Repository

interface OpenTopicByCategoryRepository : Repository<OpenTopicByCategory, Long> {
    fun saveAll(topics: Iterable<OpenTopicByCategory>)

    @Query("select t from OpenTopicByCategory t " +
            "where year(t.date) = year(current_date) " +
            "and month(t.date) = month(current_date)")
    fun findAllByCurrentMonth(): List<OpenTopicByCategory>
}